### DISCUSSION TITLE
Programmatically create collections etc.

### DISCUSSION THREAD

**Discussion by mgholam** [pocketbase/pocketbase#890] [Q&A]
Is there a way to programmatically create collections and generally script the creation process so we can put in source control?

**Comment by ganigeorgiev** [comment]
This depends on what you are trying to do. If you want to just commit your collections in git, currently you have 2 options:

- Export them to json file from the Admin UI in "Settings > Export Collections". Later they can be reimported from "Settings > Import Collections".

- Create a DB migration. This works only if you use [PocketBase as framework](https://pocketbase.io/docs/use-as-framework/). There is also an automated migration command that will export a snapshot of your current collections configuration in a migration file - https://pocketbase.io/docs/db-migrations/#collections-migration.

There will be more documentation on this with the v0.8.0 site docs update, but if you want to create collections manually via code, you could do something like:

```go
collection := &models.Collection{
    Name:       "myNewCollection",
    Type:       models.CollectionTypeBase, // this is the default
    ViewRule:   types.Pointer(""),
    CreateRule: types.Pointer("@request.auth.id != ''"),
    DeleteRule: nil, // only admins will be able to access
    Schema: schema.NewSchema(
        &schema.SchemaField{
            Name:     "title",
            Type:     schema.FieldTypeText,
            Unique:   true,
            Required: true,
            Options:  &schema.TextOptions{},
        },
        // ... other fields
    ),
}

err := app.Dao().SaveCollection(collection)
```

---

The current export/import json flow when the prebuilt PocketBase executable is used is not ideal and will be fixed in the near future. There  are several discussions on this but the most recent one I think is this one - https://github.com/pocketbase/pocketbase/discussions/489.
